rocketmq-spring-boot-starter/src/main/java/org/apache/rocketmq/spring/starter/config/TransactionHandlerRegistry.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.rocketmq.spring.starter.config;
+
+import io.netty.util.internal.ConcurrentSet;
+import org.apache.rocketmq.client.exception.MQClientException;
+import org.apache.rocketmq.spring.starter.core.RocketMQTemplate;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+
+public class TransactionHandlerRegistry implements DisposableBean {
+  @Autowired
+  private RocketMQTemplate rocketMQTemplate;
+
+  private final Set<String> listenerContainers = new ConcurrentSet<>();
+
+  public TransactionHandlerRegistry() {
+  }
+
+  public Collection<String> getAllTrans() {
+    return Collections.unmodifiableSet(listenerContainers);
+  }
+
+  @Override
+  public void destroy() throws Exception {
+    listenerContainers.clear();
+  }
+
+  public void registerTransactionHandler(TransactionHandler handler) throws MQClientException {
+    if (listenerContainers.contains(handler.getName()))
```

Comment:
Consider the case of the many containers existence?

Reply:
a listener needs a container to hold. In case of there are multiple checklisteners are defined in a JVM,  that need multiple containers.